lib/serverless.js

```diff
@@ -215,6 +220,49 @@ class Serverless {
   logDeprecation(code, message) {
     return this._logDeprecation(`EXT_${ensureString(code)}`, ensureString(message));
   }
+
+  extendConfiguration(configurationPathKeys, value) {
+    ensureArray(configurationPathKeys, {
+      ensureItem: ensureString,
+    });
+    if (configurationPathKeys.length < 1) {
+      throw new ServerlessError(
+        'ConfigurationPathKeys needs to contain at least one element.',
+        'INVALID_EXTEND_AFTER_INIT'
+      );
+    }
+
+    if (!this.isConfigurationExtendable) {
+      throw new ServerlessError(
+        'ExtendConfiguration cannot be used after init.',
+        'INVALID_EXTEND_AFTER_INIT'
+      );
+    }
+    try {
+      value = JSON.parse(JSON.stringify(value));
+    } catch (error) {
+      throw new ServerlessError(
+        'ExtendConfiguration called with invalid data. Value is not json-serializable. Error: ${error}',
+        'INVALID_EXTEND_VALUE'
+      );
+    }
+
+    const configurationPath = configurationPathKeys.join('.');
+    _.set(this.configurationInput, configurationPath, value);
+
+    const metaPathPrefix = configurationPathKeys.join('\0');
+    for (const key of this.variablesMeta.keys()) {
+      if (key.startsWith(metaPathPrefix)) {
```

Comment:
We should search for one that equals `metaPathPrefix` and ones that start with `metaPathPrefix + '\0'`

Reply:
Included that case in the tests and it worked flawlessly.

Reply:
The problem is that it's too eager for paths such as `test.foo` it will pick `test.foobar` which shouldn't be the case

Reply:
Wow didn't think of that. Will fix it and add it to testing.